wizard/src/components/mapping_table_header.tsx

```diff
@@ -18,36 +18,65 @@
  * Component for showing the header of the table preview in the mapping section.
  */
 
-import React from "react";
+import React, { useEffect, useState } from "react";
+import { Input } from "reactstrap";
 
-import { Column, MAPPED_THING_NAMES, MappedThing, MappingType } from "../types";
+import { MAPPED_THING_NAMES, MappedThing, MappingType } from "../types";
 import { ColumnInfo } from "./mapping_section";
 
+interface EditHeaderState {
+  colIdx: number;
+  header: string;
+}
 interface MappingTableHeaderProps {
-  orderedColumns: Column[];
-  columnInfo: Record<string, ColumnInfo>;
+  columnInfo: Map<number, ColumnInfo>;
   onColumnSelected: (colIdx: number) => void;
   highlightedColumnRef: React.RefObject<HTMLTableHeaderCellElement>;
   selectedColumn: number;
+  onColumnUpdated: (colIdx: number, column: ColumnInfo) => void;
 }
 
 export function MappingTableHeader(
   props: MappingTableHeaderProps
 ): JSX.Element {
+  const [editHeaderState, setEditHeaderState] = useState<EditHeaderState>({
+    colIdx: -1,
+    header: "",
+  });
+  const orderedColumns = Array.from(props.columnInfo.keys()).sort((a, b) => {
+    if (a > b) {
+      return 1;
```

Comment:
nit: Trying to understand why the default sort() function or sort().reverse() won't work given these are numbers?

Reply:
the default sort function sorts alphabetically so the sorted order would be like [1, 100, 2, 250, 26, ..]

Reply:
since these are numbers, you could also do `.sort((a,b)=> a-b)`

Reply:
done